packages/@sanity/block-tools/src/HtmlDeserializer/rules/html.ts

```diff
@@ -14,17 +14,24 @@ import {
 } from '../../constants'
 import {BlockEnabledFeatures, DeserializerRule} from '../../types'
 
-export function resolveListItem(listNodeTagName: string): string {
+export function resolveListItem(
+  listNodeTagName: string,
+  enabledListTypes: string[],
+): string | undefined {
   let listStyle
   switch (listNodeTagName) {
     case 'ul':
-      listStyle = 'bullet'
+      if (enabledListTypes.includes('bullet')) {
+        listStyle = 'bullet'
+      }
       break
     case 'ol':
-      listStyle = 'number'
+      if (enabledListTypes.includes('number')) {
+        listStyle = 'number'
+      }
       break
     default:
-      listStyle = 'bullet'
+      listStyle = undefined
   }
   return listStyle
```

Comment:
This function seems a bit cumbersome to me after the conditionals were added, maybe we could simplify it? 
```ts
  if (listNodeTagName === 'ul' && enabledListTypes.includes('bullet')) {
    return 'bullet'
  }
  if (listNodeTagName === 'ol' && enabledListTypes.includes('number')) {
    return 'number'
  }
  return undefined
```

Reply:
Better! I've changed it, tnx!